### PR TITLE
Support maxval() intrinsic

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
fixes https://github.com/lfortran/lfortran/issues/1863

### CODE REVIEW

File: src/libasr/codegen/asr_to_llvm.cpp [hunk 1]
```diff
@@ -5587,15 +5587,10 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
...
-            this->visit_expr_wrapper(x.m_body, true);
-            then_val = tmp;
-        }, [=, &else_val]() {
-            this->visit_expr_wrapper(x.m_orelse, true);
-            else_val = tmp;
-        });
+        this->visit_expr_wrapper(x.m_body, true);
+        llvm::Value *then_val = tmp;
+        this->visit_expr_wrapper(x.m_orelse, true);
+        llvm::Value *else_val = tmp;
```
Comment (certik) [inline src/libasr/codegen/asr_to_llvm.cpp:5593]: Test this change in LPython, I think the new implementation is not correct.

**Comment by certik** [comment]
I think this looks great. I am worried about https://github.com/lfortran/lfortran/pull/1901/commits/d9b857c9ebb3253cd91f184c0a56f9a5ffd6f154, can you explain why the old implementation is not correct and the new one is? I think we have some tests for this in LPython.

File: src/libasr/codegen/asr_to_llvm.cpp (hunk 1 above)
Comment (certik) [reply src/libasr/codegen/asr_to_llvm.cpp:5593]: It looks like this change is correct, see https://github.com/lcompilers/lpython/pull/2070.

**Review by certik — APPROVED** [review]
I think this is good, thanks for this!

File: integration_tests/arrays_intrin_04.f90 [hunk 2]
```diff
@@ -0,0 +1,16 @@
+program arrays_intrin_04
+    integer :: i, j
+    real(8) :: x(3, 5), xdiff
+
+
+    do i = 1, 3
+        do j = 1, 5
+            x(i, j) = (i + j) / 7
```
Comment (certik) [inline integration_tests/arrays_intrin_04.f90:8]: This is integer division...

File: integration_tests/arrays_intrin_04.f90 [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        end do
+    end do
+
+    xdiff = abs(maxval(x) - (8 / 7))
```
Comment (certik) [inline integration_tests/arrays_intrin_04.f90:12]: Integer division in 8/7.

**Comment by certik** [comment]
The integer division bugs are fixed in https://github.com/lfortran/lfortran/pull/1903.

File: integration_tests/arrays_intrin_04.f90 (hunk 3 above)
Comment (Shaikh-Ubaid) [reply integration_tests/arrays_intrin_04.f90:12]: Got it.  Also I think the double precision `_dp` is missing. I will take care about these in future. Thank you so much for the guidance.

Comment (certik) [reply integration_tests/arrays_intrin_04.f90:12]: Yes. We should probably warn when integer division is used, it's a common source of mistakes. We'll play with such warnings once we can compile lots of codes, and we'll test the warning on the codes to ensure we don't get too many false positives (those can be silenced by `int(8) / 7` or something like that).

**Comment by HarshitaKalani** [comment]
Would it be fine if I try doing the same for minval() following this?

**Comment by certik** [comment]
@HarshitaKalani yes, please do!

**Pull request merged**